packages/base/card-api.gts

```diff
@@ -2383,6 +2417,13 @@ async function _updateFromSerialized<T extends BaseDefConstructor>(
         );
       }
       let deserialized = getDataBucket(instance);
+
+      // Before updating field's value, we also have to make sure
+      // the subscribers also subscribes to a new value.
+      let existingValue = deserialized.get(fieldName as string);
+      if (isCardOrField(existingValue) && isCardOrField(value)) {
```

Comment:
lets also add to the `if` statement that `&& existingValue !== value`. it doesn't make sense to migrate subscribers to the same instance.